nx/test/nx/defn_test.exs

```diff
@@ -243,6 +243,26 @@ defmodule Nx.DefnTest do
   describe "unary ops" do
     defn exp(t), do: Nx.exp(t)
 
+    defn unary_plus_minus_guards(t, opts \\ []) do
+      case opts[:value] do
+        value when value == -2 ->
+          Nx.add(t, value)
```

Comment:
Make this return -1 or 1 in order to simplify the tests. See the tests I have added for `case` itself.